modules/renter/contractor/contractor_test.go

```diff
@@ -130,28 +132,128 @@ type stubHostDB struct{}
 func (stubHostDB) Host(modules.NetAddress) (h modules.HostDBEntry, ok bool)         { return }
 func (stubHostDB) RandomHosts(int, []modules.NetAddress) (hs []modules.HostDBEntry) { return }
 
-// TestSetAllowance tests the SetAllowance method.
-func TestSetAllowance(t *testing.T) {
-	c := &Contractor{
-		// an empty hostDB ensures that calls to formContracts will always fail
-		hdb: stubHostDB{},
+// TestIntegrationSetAllowance tests the SetAllowance method.
+func TestIntegrationSetAllowance(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	// create testing trio
+	h, c, m, err := newTestingTrio("TestIntegrationSetAllowance")
+	if err != nil {
+		t.Fatal(err)
 	}
 
 	// bad args
-	err := c.SetAllowance(modules.Allowance{Funds: types.NewCurrency64(1), Period: 0, Hosts: 3})
+	var a modules.Allowance
+	err = c.SetAllowance(a)
 	if err == nil {
 		t.Error("expected error, got nil")
 	}
-	err = c.SetAllowance(modules.Allowance{Funds: types.NewCurrency64(1), Period: 2, Hosts: 0})
+	a.Hosts = 1
+	err = c.SetAllowance(a)
 	if err == nil {
 		t.Error("expected error, got nil")
 	}
-
-	// formContracts should fail (no hosts)
-	err = c.SetAllowance(modules.Allowance{Funds: types.NewCurrency64(1), Period: 2, Hosts: 3})
+	a.Period = 20
+	err = c.SetAllowance(a)
+	if err == nil {
+		t.Error("expected error, got nil")
+	}
+	a.RenewWindow = 20
+	err = c.SetAllowance(a)
 	if err == nil {
 		t.Error("expected error, got nil")
 	}
+
+	// reasonable values; should succeed
+	a.Funds = types.SiacoinPrecision.Mul64(100)
+	a.RenewWindow = 10
+	err = c.SetAllowance(a)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(c.contracts) != 1 {
+		t.Error("expected 1 contract, got", len(c.contracts))
+	}
+
+	// set same allowance; should no-op
+	err = c.SetAllowance(a)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(c.contracts) != 1 {
+		t.Fatal("expected 1 contract, got", len(c.contracts))
+	}
+
+	// reannounce host on different IP (easier than creating a new host)
```

Comment:
This test will stop working once we switch to identifying hosts by public key. Probably worthwhile to make a `addiitonalHost` helper function - I'm guessing that thorough testing will want to be able to spin up many hosts in multiple different tests.